File: doan/lib/models/movie.dart
class Movie {
  final int id;
  final String name;
  final String description;
  final int? duration;
  final String releaseDate;
  final String posterUrl;
  final String bannerUrl;
  final String? ageRating;
  final double voteAverage;

  Movie({
    required this.id,
    required this.name,
    required this.description,
    this.duration,
    required this.releaseDate,
    required this.posterUrl,
    required this.bannerUrl,
    this.ageRating,
    required this.voteAverage,
  });
}

File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/Models/Movie.java
package com.project.bookingmovieticketapp.Models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "movies")
@Builder
public class Movie {
    @Id
    private int id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;
    private int duration;

    @Column(name = "releasedate", nullable = false)
    private LocalDate releasedate;

    private String posterurl;
    private String bannerurl;
    private String agerating;

    @Column(name = "voteaverage", nullable = false)
    private Double voteaverage;

    private String director;
}
